e2e/suites/provider/cases/template/template.go

```diff
@@ -133,10 +135,10 @@ func genericPushSecretTemplate(f *framework.Framework) (string, func(*framework.
 			},
 		}
 		tc.VerifyPushSecretOutcome = func(sourcePs *esv1alpha1.PushSecret, pushClient esv1.SecretsClient) {
-			gomega.Eventually(func() bool {
+			Eventually(func() bool {
 				s := &esv1alpha1.PushSecret{}
-				err := tc.Framework.CRClient.Get(context.Background(), types.NamespacedName{Name: tc.PushSecret.Name, Namespace: tc.PushSecret.Namespace}, s)
-				gomega.Expect(err).ToNot(gomega.HaveOccurred())
+				err := tc.Framework.CRClient.Get(GinkgoT().Context(), types.NamespacedName{Name: tc.PushSecret.Name, Namespace: tc.PushSecret.Namespace}, s)
```

Comment:
eventually needs a WithContext argument and GinkgoT().Context() likely needs to be replaced with (ctx SpecContext) you receive from BeforeEach or other GinkgoNodes. See https://onsi.github.io/ginkgo/#interruptible-nodes-and-speccontext for reference. I believe this is a pretty big change though so this here is at least better than context.Background as it terminates on test end